pyproject.toml

```diff
@@ -1,7 +1,67 @@
+# region #### Project metadata ##########
+
+[project]
+name = "aiosmtpd"
+requires-python = ">=3.7"
+dynamic = [ "version" ]
+description = "aiosmtpd - asyncio based SMTP server"
+readme = "DESCRIPTION.rst"
+urls."Home Page" = "https://aiosmtpd.readthedocs.io/"
+urls."Bug Tracker" = "https://github.com/aio-libs/aiosmtpd/issues"
+urls."Documentation" = "https://aiosmtpd.readthedocs.io/"
+urls."Source Code" = "https://github.com/aio-libs/aiosmtpd"
+maintainers = [
+    { name = "The aiosmtpd Developers", email = "aiosmtpd@googlegroups.com"}
+]
+keywords = [ "email", "smtpd", "smtp", "async" ]
+license.file = "LICENSE"
+classifiers = [
+    "License :: OSI Approved",
+    "License :: OSI Approved :: Apache Software License",
+    "Intended Audience :: Developers",
+    "Operating System :: Microsoft :: Windows",
+    "Operating System :: POSIX :: BSD :: FreeBSD",
+    "Operating System :: POSIX :: Linux",
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: Implementation :: CPython",
+    "Programming Language :: Python :: Implementation :: PyPy",
+    "Topic :: Communications :: Email :: Mail Transport Agents",
+    "Framework :: AsyncIO",
+]
+dependencies = [
+    "atpublic",
+    "attrs",
+    "typing-extensions ; python_version < '3.8'",
```

Comment:
Are typing extensions require to run aiosmtpd?

Reply:
iirc it's to implement Protocol, I think that's in the Tproxy code.

Not really sure if that's actually needed to just run ...

Reply:
Assuming that it's not needed for running we can put it in an `extras_require` (yeah, not extra_requires) section. I'm not sure what the precise form is for that, tho.

Reply:
If that were true, it should be here, not in extras. FWIW it's only used in tests: https://github.com/aio-libs/aiosmtpd/blob/abadbd41dabb81eda4fbb196e1dc590bcc908648/aiosmtpd/tests/test_handlers.py#L30. Judging by the lack of it in runtime code, it might be fine to drop it from the tests and deps.